packages/mantine/src/BlockNoteView.tsx

```diff
@@ -74,24 +74,28 @@ export const BlockNoteView = <
     [defaultColorScheme, theme]
   );
 
+  const mantineCssVariablesRef = useRef<HTMLDivElement>(null);
+
   return (
     <ComponentsContext.Provider value={components}>
       {/* `cssVariablesSelector` scopes Mantine CSS variables to only the editor, */}
       {/* as proposed here:  https://github.com/orgs/mantinedev/discussions/5685 */}
-      <MantineProvider
-        theme={mantineTheme}
-        cssVariablesSelector=".bn-mantine"
-        // This gets the element to set `data-mantine-color-scheme` on. Since we
-        // don't need this attribute (we use our own theming API), we return
-        // undefined here.
-        getRootElement={() => undefined}>
-        <BlockNoteViewRaw
-          className={mergeCSSClasses("bn-mantine", className || "")}
-          theme={typeof theme === "object" ? undefined : theme}
-          {...rest}
-          ref={ref}
-        />
-      </MantineProvider>
+      <div className={"bn-mantine-css-variables"} ref={mantineCssVariablesRef}>
+        <MantineProvider
+          theme={mantineTheme}
+          cssVariablesSelector=".bn-mantine-css-variables"
+          // This gets the element to set `data-mantine-color-scheme` on. Since we
+          // don't need this attribute (we use our own theming API), we return
+          // undefined here.
+          getRootElement={() => mantineCssVariablesRef.current || undefined}>
```

Comment:
Little unclear, the comment says it isn't needed, but we still are giving it the ref?

Reply:
Ah I see now, this is the change you wanna make, well the comment should be updated too.

But, am I crazy that all this is doing is setting that `data-mantine-color-scheme` can't we just set that attribute ourselves? Or maybe we don't know the color scheme?

Reason being that React normally doesn't like elements being "touched" by plain JS and on a re-render will just remove the extra attribute as though it were never there. Maybe that is the root cause of this issue?

But, IDK. I've never used Mantine nor understand the internals here

Reply:
Ah sorry, that's my bad I missed changing the comment.

I think when I wrote this comment originally I misunderstood what `cssVariablesSelector` and `getRootElement` were for. To clarify, `cssVariablesSelector` is for additional variables from the Mantine theme, and `getRootElement` is the base set of variables to make Mantine components work.

Either way, seems like it's more than just `data-mantine-color-scheme` being set - also CSS variables. Sorry again for the misleading comment.